### ISSUE TITLE
Discrepancy between measured/reported filesystem `free` space and actual usable `available` space on `ext4` when running as non-proviledged user

### ISSUE THREAD

**Issue by Jmennius** [chrislusf/weed-fs#5991]
There is a discrepancy between measured/reported filesystem `free` space and actual usable `available` space on `ext4` if running as a regular user.
Seaweedfs will report a certain size as disk `Free` space. But it cannot necessarily use all of it making this estimation wrong on `ext4` because of 'reserved blocks' feature.
Logic based on `minFreeSpace` can be broken in this case.

**System Setup**
- docker container: `server -s3 -s3.config ./s3-config.json -volume.minFreeSpace=10 -volume.max=0`, /data mounted to ext4 filesystem on the host (that has reserved blocks - default behavior)
- Ubuntu 20.04
- `version 30GB 3.64 b74e8082b linux amd64`

**Expected behavior**
If seaweedfs estimates, or shows a certain number as 'free' space - it should be able to use this space.

**Additional context**
`Ext` filesystems have a feature where it reserves a number of blocks for a superuser, this is by default 5% of filesystem size.
`statfs(2)` reports two numbers: `f_bfree` (Free blocks in filesystem) and `f_bavail` (Free blocks available to unprivileged user), usually `f_bavail <= f_bfree`.
Seaweedfs uses `f_bfree` for it's reporting and disk estimations, but it can't always use it making the estimation wrong and misleading. Running as a regular user vs root influences this.

`df` reports 'available'. `stat -f /path/to/mount` reports both free and available blocks. `tune2fs -l /dev/partition` can be used to check both free and reserved blocks.

**Comment by chrislusf** [comment]
Thanks for the detailed context! Could you fix it with a PR in https://github.com/seaweedfs/seaweedfs/blob/master/weed/mount/weedfs_stats.go ?

**Comment by Jmennius** [comment]
> Thanks for the detailed context! Could you fix it with a PR in https://github.com/seaweedfs/seaweedfs/blob/master/weed/mount/weedfs_stats.go ?

Yes, I want to, but thought that maybe it's better to discuss it first - maybe there are other consequences (should we change naming anywhere accordingly?).
I'm also thinking that the relevant value is 'free' for root and 'available' for non-root - need to verify that...
